### PR TITLE
Optimize React for production

### PR DESCRIPTION
**Pull request by blgm**
- sets `process.env.NODE_ENV = 'production'` using `envify`
- eliminates the (now dead) React development code with `uglifyify`
- reduces the bundle size from `1028132` bytes to `655814` (36%)
- based on guidance on how to use React with browserify: https://reactjs.org/docs/optimizing-performance.html#browserify
- fixes https://github.com/alanshaw/david-www/issues/390

What I don't like about these changes:
- there is now some minification in a build phase (it should all be in the `minify` phase)
- you have to change `package.json` to get a development build (if we knew that `NODE_ENV` was always set to `production` on the production server, then we could allow `envify` to pick it up from the environment, allowing development machines to have a development version of React)
- I haven't updated `package-lock.json` as both `.travis.yml` and `package.json` suggest this is a node 6 project, which does not support `package-lock.json` by default

I'm happy to try and resolve these with some guidance as to what is preferred.

Testing:
- unit tests pass locally and [on Travis](https://travis-ci.org/alanshaw/david-www/builds/345697545)
- can run server on my local system
- I was able to reproduce the issue, and show that these changes fixed it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "start": "node dist/main",
     "build": "run-p build:**",
     "build:js:server": "cross-env BABEL_ENV=server babel src --out-dir dist",
-    "build:js:client": "browserify -g envify src/ui/main.jsx -o public/bundle.js",
+    "build:js:client": "browserify -g [ envify purge --NODE_ENV production ] -g uglifyify src/ui/main.jsx -o public/bundle.js",
```
Comment (alanshaw) [inline package.json]: Do we need to add `envify` to `devDependencies`?

Comment (alanshaw) [reply package.json]: Do we need `uglifyify`?  The `minify:js` script does that

**Comment by alanshaw** [comment]
The [infrastructure project is here](https://github.com/alanshaw/david-www-infrastructure), I need to upgrade it to add letsencrypt support this week so I can also do the work to ensure NODE_ENV set to production in the same pass so lets leave it out of the package.json.

I'd appreciate it if you did also submit an updated `package-lock.json` I want to use a more recent Node.js in production so we'll need it, and it is in the repo now so we should at least keep it up to date.

**Comment by alanshaw** [comment]
Also, thanks ✨ @blgm ✨

**Comment by blgm** [comment]
@alanshaw, I've pushed some more changes which hopefully address your comments.

**Pull request merged**